odxtools/cli/compare.py

```diff
@@ -467,11 +472,11 @@ def compare_diagnostic_layers(self, dl1: DiagLayer, dl2: DiagLayer) -> dict:
                         # add parameters which have been changed (type: String)
                         service_dict["changed_parameters_of_service"][
                             1].append(  # type: ignore[union-attr]
-                                detailed_information[1])
+                                detailed_information[1])  # type: ignore[arg-type]
```

Comment:
why?

Reply:
without it, strict mypy produces
```
odxtools/cli/compare.py:475: error: Argument 1 to "append" of "list" has incompatible type "dict[str, str | str | list[DiagService] | list[DiagService] | list[list[str | DiagService]] | list[list[str | DiagService]]]"; expected "str | DiagService"  [arg-type]
```

The problem is that duck-typing is used extensively in the compare tool. (I have it on my list for a large-ish  refactoring...)